### PR TITLE
[CORE-1809] Validate effective specs

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/server/pps/server/defaults.go [hunk 1]
```diff
@@ -322,3 +352,42 @@ func unmarshalJSON(s string, v any) error {
...
 	return errors.Wrapf(d.Decode(v), "could not unmarshal %q as JSON", s)
 }
+
+func staticallyValidate(req *pps.CreatePipelineRequest) error {
+	// TODO(msteffen) eventually TFJob and Transform will be alternatives, but
+	// currently TFJob isn't supported
+	if req.TfJob != nil {
+		return errors.New("embedding TFJobs in pipelines is not supported yet")
+	}
+	if !(req.ReprocessSpec == "" || req.ReprocessSpec == client.ReprocessSpecUntilSuccess && req.ReprocessSpec != client.ReprocessSpecEveryJob) {
```
Comment (acohen4) [inline src/server/pps/server/defaults.go:362]: shouldn't the last conditional be `==` instead of `!=`?

**Review by acohen4 — COMMENTED** [review]
just one comment

File: src/server/pps/server/defaults.go (hunk 1 above)
Comment (robert-uhl) [reply src/server/pps/server/defaults.go:362]: Yeah, and a `||` too.

I fixed a test for the permissible values, as well.

**Review by acohen4 — APPROVED** [review]
LGTM

**Pull request merged**